### ISSUE TITLE
Complete remaining feature work for removing plugin call order preference

### ISSUE THREAD

**Issue by DragaDoncila** [napari/napari#4390]
## 🧰 Task
As originally proposed in #4000, we would like to remove the plugin call order preference left over from the original `napari-plugin-engine`.

In order to remove this behavior we must make sure that GUI users and command line users both have reasonable ability to discern when multiple plugins are available for their file, what those plugins are and how they can save preferences for common filetypes.

This issue documents the remaining tasks/features which are must haves and nice to haves before we can remove the call order behavior, partially for my own sanity, but also to make sure everyone is on board. The must haves are written in order of priority and the order in which we'll be working on them. 

### Must Have

- [x] 1. Consolidate behavior between GUI and command line (PR #4347)
- [x] 2. Provide GUI method for associating extensions to readers (Issue #4146)
- [ ] 3. Update our preference saving logic to support associating preferences for directories with extensions
- [x] 4. @nclack @tlambert03 I believe there's a final piece of the puzzle on the napari side to support silent conversion of `npe1` plugins to `npe2`?

### Nice to Have
- [ ] Use reader contribution title to associate plugins, rather than plugin name (Issue #4391)
- [ ] Decide what we want to do about `filename_pattern` matching and `directories` and implement (see [this zulip](https://napari.zulipchat.com/#narrow/stream/309872-plugins/topic/directory.20name.20patterns.20in.20reader.20plugins.3F) discussion and [this issue](https://github.com/napari/npe2/issues/155))
- [ ] Documentation on how opening behavior works, how the original one worked (in case people get confused) and how we support various use cases e.g. directories, extensions, stacking, etc.
- [x] Shortcut to force dialog open (Issue #4388)
- [x] Special casing `builtins` when just one "third-party" plugin is available (Issue #4389)

**Comment by psobolewskiPhD** [comment]
@DragaDoncila 
How does this issue jive with removing npe1?

**Comment by DragaDoncila** [comment]
@psobolewskiPhD thanks for flagging. Certainly the call order preference dialog has been removed, but there's still a couple of items here that would be useful to have. Afaict they pretty much all have associated issues though, so we can probably close this.
